### PR TITLE
Skip progress indicator test when viewer is not shown

### LINKED ISSUE #3061: Progress test breaks local testing
## 🐛 Bug

this line, added in #2931 breaks local tests when you just run `pytest`, since it's expecting the viewer to be shown:

https://github.com/napari/napari/blob/712a96e27fc901d052abb37202d15ccf285b52f3/napari/_qt/_tests/test_progress.py#L161

(I'm guessing it worked for @DragaDoncila locally because that particular module shows things for linux as well?)

## To Reproduce

```
pytest napari/_qt/_tests/test_progress.py
```

## Expected behavior

should be able to run tests locally without setting CI=1

### PR DESCRIPTION
**Pull request by DragaDoncila**
# Description
This PR skips the test checking whether the progress indicator is shown if the viewer is currently not being shown.

The test relies on checking whether the progress indicator is visible, which will always be false if the viewer is not visible.
This should fix local tests for non-linux distributions. @tlambert03 can you confirm if this is the right approach?

## Type of change
- [x] Bug-fix (non-breaking change which fixes an issue)

# References
Closes #3061

# How has this been tested?
All tests still pass with my change but will need someone to confirm who's not running on linux. 

## Final checklist:
- [ ] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/docs/dev/guides/translations.html).

### CODE REVIEW

**Review by sofroniewn — APPROVED** [review]
LGTM! Will wait for @tlambert03 to weigh in to confirm before merge

**Review by tlambert03 — APPROVED** [review]
Thanks @DragaDoncila!

**Pull request merged**
